### ISSUE TITLE
Typed percentage is gone when using NTI?

### ISSUE THREAD

**Issue by vobruba-martin** [google/closure-compiler#2738]
@dimvar My build scripts stopped reporting typed percentage after 52194cf080a8363b250b8f6c0fcb88c11e639122. Is it intended?

**Comment by MatrixFrog** [comment]
I don't know if I'd consider that a very important feature but it is potentially good information to have. How do you use it (for instance, do you have a goal of keeping it above 90%)?

**Comment by vobruba-martin** [comment]
My goal is keeping it as high as possible. My largest project (currently 50 000+ lines) is having 99.97% typed. I've found the percentage very useful in some scenarios where Closure Compiler didn't emit any warning but I could see the percentage drop. From what I remember:

### Undefined property
Very common for me. This usually means that I have misspelled property name.

```javascript
/**
 * @record
 * @struct
 */
function MyRecord() {};

/** @type {boolean} */
MyRecord.prototype.prop;

/**
 * @param {!MyRecord} record
 */
function DoSomething(record) {
	if (record.porp) {    // property name misspelled
		console.warn('!!!');
	}
};
```

### Missing annotation
```javascript
/**
 * @return {void}
 */
function DoSomething_1(prop1) {
	DoSomething_2(prop1);    // missing prop1 annotation, type safety is broken
};

/**
 * @param {string} prop1
 * @return {void}
 */
function DoSomething_2(prop1) {
	console.warn(prop1);    // prop1 is number!
};

/**
 * @param {number} prop1
 * @return {void}
 */
function DoSomething_3(prop1) {
	DoSomething_1(prop1);
};

DoSomething_3(1);
```

Or

```javascript
/**
 * @param {string=} prop1
 */
function ReturnTypeNotDefined(prop1) {
	return true;
};

/**
 * @return {void}
 */
function DoSomething() {
	if (ReturnTypeNotDefined()) {    // missing @return annotation
		console.warn('!!!');
	}
};
```

**Comment by MatrixFrog** [comment]
The conformance framework is a better way of finding some of those things but it's more work to get it up and running, especially on a large existing codebase. I don't know how much work it would be to add the percentage on NTI but I think it's worth considering. In fact we could even get more specific and emit a per-file report (80% of the code in file1.js is typed, 90% in file2.js is typed...) and then users could decide they want a higher threshold for some files than others.
